classes/search/ArticleSearchDAO.inc.php

```diff
@@ -88,7 +90,7 @@ function getPhraseResults($journal, $phrase, $publishedFrom = null, $publishedTo
 				i.published = 1 AND ' . $sqlWhere . '
 			GROUP BY o.submission_id
 			ORDER BY count DESC
-			LIMIT ' . $limit,
+			' . sprintf($isSqlServer ? 'OFFSET 0 ROWS FETCH NEXT %d ROWS ONLY' : 'LIMIT %d', $limit),
```

Comment:
ADODB provides tools for abstracting the SQL details for making ranged queries -- but what's missing is a function in `lib/pkp/classes/db/DAO.inc.php` that allows these queries to be made with caching enabled. Essentially a combination of `DAO::retrieveCached` and `DAO::retrieveLimit`. I think it makes sense to add that function, then let ADODB handle the range part of the SQL formulation. Let me know if I can help with this.

Reply:
I will take a look at those functions and see if I can understand it and make the SQL valid for every databases.